resources/testharness.js

```diff
@@ -1741,6 +1741,12 @@ policies and contribution forms [3].
             }
         };
 
+        if (self.Promise) {
+            this.done = new Promise(function(resolve) {
```

Comment:
Is there a reason to create this property, or is it just because that's the easiest way to pass the promise to the caller?

Reply:
Yeah, this is purely so it can be returned in https://github.com/w3c/web-platform-tests/pull/10348/files#diff-d9d64f7c061650556b18794cc9719152R2159.